web3/providers/eth_tester/middleware.py

```diff
@@ -251,23 +253,51 @@ def is_named_block(value):
 })
 
 
+def guess_from(web3, transaction):
+    if web3.eth.coinbase:
+        return web3.eth.coinbase
```

Comment:
Each of these if/return statements causes a double RPC call.  It would be better to write this as

```
try:
    return web3.eth.coinbase
except Whatever:
    pass

try:
    return web3.eth.accounts[0]
except Whatever:
    pass

return None
```

Reply:
In the first case I'm checking if `coinbase` is not None, which wouldn't raise anything, but I can still do:

```py
coinbase = web3.eth.coinbase
if coinbase is not None:
    return coinbase
```

Reply:
Consider my code above to be psuedo code.  Change it out for whatever matches your desired logic (while still reducing API calls).